src/pages/index.js

```diff
@@ -32,71 +86,150 @@ const cardPopOut = new PopupWithImage({
 
 cardPopOut.setEventListeners();
 
-function createCard(item) {
-  const card = new Card(item, "#localeCard", openModalImage);
-  return card.generateCard();
+const deleteCardModal = new PopupWithDelete({
+  popupSelector: ".modal_type_delete-image",
+  handleBtnClick: () => {
+    api
+      .deleteCard(cardToDelete.data._id)
+      .then(() => cardToDelete.deleteCard())
+      .then(() => deleteCardModal.close())
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => deleteCardModal.renderSaving(false));
+  },
+  buttonText: "Yes",
+});
+
+deleteCardModal.setEventListeners();
+
+function handleDeleteClick(card) {
+  console.log(card);
+  deleteCardModal.open();
+  cardToDelete = card;
+  return cardToDelete;
 }
 
-const firstCards = new Section(
-  { items: initialCards, renderer: createCard },
-  ".elements"
-);
+function handleLike(card) {
+  cardID = card.getID();
+  if (card.data.isLiked == false) {
+    api
+      .addLike(cardID)
+      .then((card.data.isLiked = true))
```

Comment:
`      .then((card.data.isLiked = true))
`

In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 


```js
      .then(() => card.data.isLiked = true)
      .then(() => card.toggleHeart())

```

If you have 2 `thens` in a row, you can combine them together


```js
      .then(() => {
        card.data.isLiked = true;
        card.toggleHeart();
      })
      .catch((err) => {

```

**Please, correct it everywhere**